web_widget_timepicker/static/src/js/web_widget_timepicker.js

```diff
@@ -0,0 +1,115 @@
+            if(!this.get("effective_readonly")) {
+                var custom_options;
+
+                if( this.node.attrs['data-options'] !== undefined && this.node.attrs['data-options'].length > 0) {
+                    // First try to use jquery data function to create object
+                    custom_options = $(this.node).data('options');
+
+                    if(typeof custom_options !== 'object') {
+                        // No garantee that the input data-options string is valid JSON format so try to cleanup
+                        custom_options = JSON.parse(cleanup_str2json(this.node.attrs['data-options']));    
+                    }
+                }
+
+                if(typeof custom_options === 'object') {
+                    this.$el.find('input').timepicker($.extend({}, this.options, custom_options )); 
+                } else {
+                    this.$el.find('input').timepicker(this.options);                    
+                }
+
+                this.setupFocus(this.$('input'));                   
+            }   
+        },
+        is_syntax_valid: function() {
+            if (!this.get("effective_readonly") && this.$("input").size() > 0) {
+                try {
+                    this.parse_value(this.$('input').val(),'');
+                    return true;
+                } catch(e) {
+                    return false;
+                }
+            }
+            return true;
+        },
+        is_false: function() {
+            return this.get('value') === '' || this._super();
+        },
+        focus: function() {
+            var input = this.$('input:first')[0];
+            return input ? input.focus() : false;
+        },
+        set_dimensions: function (height, width) {
+            this._super(height, width);
+            this.$('input').css({
+                height: height,
+                width: width
+            });
+        },                        
+        store_dom_value: function () {
+            if (!this.get('effective_readonly')) {
+                this.internal_set_value(
+                    this.parse_value(
+                        this.$('input').val(),''));
+            }
+        },
+        parse_value: function(val, def) {
+            return formats.parse_value(val, {"widget": this.internal_format}, def);  
+        },
+        format_value: function(val, def) {
+            return formats.format_value(val, {"widget": this.internal_format}, def);
+        },
+        render_value: function() {
+            var show_value = this.format_value(this.get('value'),'');             
+
+            if (!this.get("effective_readonly")) {
+                this.$input = this.$el.find('input');
+                this.$input.val(show_value);
+            } else {
+                this.$(".oe_form_time_content").text(show_value);
+            }
+        },
+    });
+
+    core.form_widget_registry.add('timepicker', TimePicker);
+
+    return TimePicker;

```

Comment:
make it a habit to return an object with memebers pointing to you references. That's for future extensibility, if you release this to the wild, everyone uses it and expects this one object, you'll break everything if for whatever reason you want to offer a second class. Not probable with this module, but a good pattern to follow.